### ISSUE TITLE
Angular Frontend & Dataiku

### ISSUE THREAD

**Issue by sabyasachisabs** [dataiku/dss-code-samples#4]
Team,
We have a frontend built on angular (nginx node) and a backend api node (hosted on AWS).
Wanted to know if we can display the angular frontend to dataiku webapp and make the dataiku webapp interactive. What do you guys suggest?
Thanks,
Sabs

**Comment by alexcombessie** [comment]
Hi,

**TL;DR:** I would advise keeping your existing setup: Angular frontend + DSS API node for backend. That is a very sound setup for production and external deployment 😄 

**Longer version:** DSS webapps (hosted on Design or Automation nodes) can be developed using 3 frameworks: Flask, Bokeh, Shiny. In your case, you would need to rewrite your app from Angular to these frameworks. That rework is not really justified since you have a working setup. In addition to these, DSS webapps can only be used for internal deployment: users must be logged in DSS. If you plan to expose your webapp externally to non-DSS users, then it makes sense to keep the frontend external and to only use DSS as a real-time backend (API node).

FYI this is a similar setup to the one we demonstrated in this article: https://blog.dataiku.com/predicting-taxi-fares-in-new-york-using-machine-learning-in-real-time
Webapp: https://taxifare.dss-demo.dataiku.com/

For the future, feel free to ask your questions on our (new!) community website: https://community.dataiku.com/. I am sure your questions can benefit others 💯 

Hope it helps,

Alex
